test/lib/jdk/test/whitebox/WhiteBox.java

```diff
@@ -556,6 +560,56 @@ public void clearInlineCaches(boolean preserve_static_stubs) {
   // Force Full GC
   public native void fullGC();
 
+  // Infrastructure for waitForReferenceProcessing()
+  private Method waitForReferenceProcessingMethod = null;
+
+  private Method getWaitForReferenceProcessingMethod() {
+    Method wfrp = waitForReferenceProcessingMethod;
+    if (wfrp == null) {
```

Comment:
Racy initialization is fine, the field can be static and the last one (any racy init would all the same) wins:
```suggestion
  private static Method waitForReferenceProcessingMethod = getWaitForReferenceProcessingMethod();

  private static Method getWaitForReferenceProcessingMethod() {
```

Reply:
I can make the `Method` static, but I don't think it should be set until there's a call to `getWaitForReferenceProcessing()`, as most uses of `WhiteBox` won't need it.

Reply:
Not only will most uses of `WhiteBox` not need it, but those uses would be "broken" because they would
be forced to (without benefit) add the `@modules` declaration needed to allow that to work.